File: app/src/main/java/com/example/olesya/boardgames/AppDatabase.java
package com.example.olesya.boardgames;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.example.olesya.boardgames.interfaces.ImageHolderDao;
import com.example.olesya.boardgames.models.ImageHolder;

@Database(entities = {ImageHolder.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract ImageHolderDao getImagesDao();

    public static AppDatabase instance;
    public static AppDatabase getInstance(Context context) {
        if (instance == null) {
            //TODO: not allow
            instance = Room.databaseBuilder(context,
                    AppDatabase.class, "cards-database")
                    .allowMainThreadQueries()
                    .build();
        }

        return instance;
    }
}

File: app/src/main/java/com/example/olesya/boardgames/database/AppDatabase.kt
package com.example.olesya.boardgames.database

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context
import com.example.olesya.boardgames.entity.ImageHolder

@Database(entities = [ImageHolder::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun imagesDao(): ImageHolderDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase =
                INSTANCE ?: synchronized(this) {
                    INSTANCE ?: buildDatabase(context).also { INSTANCE = it }
                }

        private fun buildDatabase(context: Context) =
                Room.databaseBuilder(context.applicationContext,
                        AppDatabase::class.java, "cards.db")
                        .allowMainThreadQueries()
                        .build()
    }

}
